File: examples/TemplateAddUser.rb
require "hellosign-ruby-sdk"

HelloSign.configure do |config|
  # Configure HTTP basic authorization: api_key
  config.username = "YOUR_API_KEY"

  # or, configure Bearer (JWT) authorization: oauth2
  # config.access_token = "YOUR_ACCESS_TOKEN"
end

template_api = HelloSign::TemplateApi.new

data = HelloSign::TemplateAddUserRequest.new
data.email_address = "george@dropboxsign.com"

template_id = "f57db65d3f933b5316d398057a36176831451a35"

begin
  result = template_api.template_add_user(template_id, data)
  p result
rescue HelloSign::ApiError => e
  puts "Exception when calling Dropbox Sign API: #{e}"
end

File: examples/TemplateAddUser.ts
import * as HelloSign from "hellosign-sdk";

const templateApi = new HelloSign.TemplateApi();

// Configure HTTP basic authorization: api_key
templateApi.username = "YOUR_API_KEY";

// or, configure Bearer (JWT) authorization: oauth2
// templateApi.accessToken = "YOUR_ACCESS_TOKEN";

const data: HelloSign.TemplateAddUserRequest = {
  emailAddress: "george@dropboxsign.com",
};

const templateId = "f57db65d3f933b5316d398057a36176831451a35";

const result = templateApi.templateAddUser(templateId, data);
result.then(response => {
  console.log(response.body);
}).catch(error => {
  console.log("Exception when calling Dropbox Sign API:");
  console.log(error.body);
});
